src/Compilers/Core/Portable/PEWriter/PooledBlobBuilder.cs

```diff
@@ -6,29 +6,47 @@
 using System.Reflection.Metadata;
 using Microsoft.CodeAnalysis.PooledObjects;
 using Microsoft.CodeAnalysis;
+using System.Diagnostics;
 
 namespace Microsoft.Cci
 {
     internal sealed class PooledBlobBuilder : BlobBuilder, IDisposable
     {
         private const int PoolSize = 128;
-        private const int ChunkSize = 1024;
+        private const int PoolChunkSize = 1024;
 
-        private static readonly ObjectPool<PooledBlobBuilder> s_chunkPool = new ObjectPool<PooledBlobBuilder>(() => new PooledBlobBuilder(ChunkSize), PoolSize);
+        private static readonly ObjectPool<PooledBlobBuilder> s_chunkPool = new ObjectPool<PooledBlobBuilder>(() => new PooledBlobBuilder(PoolChunkSize), PoolSize);
 
         private PooledBlobBuilder(int size)
             : base(size)
         {
         }
 
-        public static PooledBlobBuilder GetInstance()
+        /// <summary>
+        /// Get a new instance of the <see cref="BlobBuilder"/> that has <see cref="BlobBuilder.ChunkCapacity"/> of
+        /// at least <see cref="PoolChunkSize"/>
+        /// </summary>
+        /// <param name="zero">When true force zero out the backing buffer</param>
+        /// <remarks>
+        /// The <paramref name="zero"/> can be removed when moving to SRM 9.0 if it contains the bug fix for
+        /// <see cref="BlobBuilder.ReserveBytes(int)"/>
+        ///
+        /// https://github.com/dotnet/runtime/issues/99244
+        /// </remarks>
+        public static PooledBlobBuilder GetInstance(bool zero = false)
         {
-            return s_chunkPool.Allocate();
+            var builder = s_chunkPool.Allocate();
+            if (zero)
+            {
+                builder.WriteBytes(0, builder.ChunkCapacity);
```

Comment:
>builder.ChunkCapacity [](http://example.com/codeflow?start=38&length=21)

Is this total memory size behind the builder?

Reply:
Yes this is the size of the underlying `byte[]` in the `BlobBuilder` instance.